test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -1177,139 +1177,6 @@ BOOST_AUTO_TEST_CASE(uncalled_blockhash)
 	)
 }
 
-BOOST_AUTO_TEST_CASE(log0)
-{
-	char const* sourceCode = R"(
-		contract test {
-			function a() public {
-				log0(bytes32(uint256(1)));
-			}
-		}
-	)";
-	ALSO_VIA_YUL(
-		DISABLE_EWASM_TESTRUN()
-
-		compileAndRun(sourceCode);
-		callContractFunction("a()");
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_CHECK_EQUAL(numLogTopics(0), 0);
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log1)
-{
-	char const* sourceCode = R"(
-		contract test {
-			function a() public {
-				log1(bytes32(uint256(1)), bytes32(uint256(2)));
-			}
-		}
-	)";
-	ALSO_VIA_YUL(
-		DISABLE_EWASM_TESTRUN()
-
-		compileAndRun(sourceCode);
-		callContractFunction("a()");
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_REQUIRE_EQUAL(numLogTopics(0), 1);
-		BOOST_CHECK_EQUAL(logTopic(0, 0), h256(u256(2)));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log2)
-{
-	char const* sourceCode = R"(
-		contract test {
-			function a() public {
-				log2(bytes32(uint256(1)), bytes32(uint256(2)), bytes32(uint256(3)));
-			}
-		}
-	)";
-	ALSO_VIA_YUL(
-		DISABLE_EWASM_TESTRUN()
-
-		compileAndRun(sourceCode);
-		callContractFunction("a()");
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_REQUIRE_EQUAL(numLogTopics(0), 2);
-		for (unsigned i = 0; i < 2; ++i)
-			BOOST_CHECK_EQUAL(logTopic(0, i), h256(u256(i + 2)));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log3)
-{
-	char const* sourceCode = R"(
-		contract test {
-			function a() public {
-				log3(bytes32(uint256(1)), bytes32(uint256(2)), bytes32(uint256(3)), bytes32(uint256(4)));
-			}
-		}
-	)";
-	ALSO_VIA_YUL(
-		DISABLE_EWASM_TESTRUN()
-
-		compileAndRun(sourceCode);
-		callContractFunction("a()");
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_REQUIRE_EQUAL(numLogTopics(0), 3);
-		for (unsigned i = 0; i < 3; ++i)
-			BOOST_CHECK_EQUAL(logTopic(0, i), h256(u256(i + 2)));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log4)
-{
-	char const* sourceCode = R"(
-		contract test {
-			function a() public {
-				log4(bytes32(uint256(1)), bytes32(uint256(2)), bytes32(uint256(3)), bytes32(uint256(4)), bytes32(uint256(5)));
-			}
-		}
-	)";
-	ALSO_VIA_YUL(
-		DISABLE_EWASM_TESTRUN()
-
-		compileAndRun(sourceCode);
-		callContractFunction("a()");
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_REQUIRE_EQUAL(numLogTopics(0), 4);
-		for (unsigned i = 0; i < 4; ++i)
-			BOOST_CHECK_EQUAL(logTopic(0, i), h256(u256(i + 2)));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log_in_constructor)
```

Comment:
Do we have this test case with `event`?

Reply:
Added a test that uses an event in the constructor: https://github.com/ethereum/solidity/pull/10092/files#diff-452c3979dc6e8b81f702d5c88aa872017141cf6a32d1f2ea839ac6750a735127R1812